### ISSUE TITLE
Plot learning curves for all Neural Network based GluonTS models like DeepAR

### ISSUE THREAD

**Issue by ealizadeh-via** [awslabs/gluon-ts#2220]
It would be very useful if we can plot learning curves for the models. Right now, I see that we can pass the validation dataset to the `.train()` method and I believe it's used for early stopping. However, plotting learning curves would also help here?

**Comment by ealizadeh-via** [comment]
So, basically what I'm asking is if there is away to get the training and validation losses from the training process?

**Comment by kashif** [comment]
on the pytorch side these are logged via pytorch lightning

**Comment by ealizadeh-via** [comment]
Thanks @kashif. 

I just found how to do that from [Trainer callbacks tutorial](https://github.com/awslabs/gluon-ts/blob/dev/docs/tutorials/advanced_topics/trainer_callbacks.md.template).

You need to pass `TrainingHistory()` callback to the `Trainer`. Then, you have access to training and validation loss. Here's an example.

```python
history = TrainingHistory()
estimator = DeepAREstimator(
    prediction_length=12,
    freq="12H",
    trainer=Trainer(
        epochs=10,
        learning_rate=1e-2,
        callbacks=[history],
    ),
)
predictor = estimator.train(training_data=ds_train, validation_data=ds_train_val)

# Retrieve losses from `history.loss_history` and `history.validation_loss_history`
train_val_loss_df = pd.DataFrame(
    {
        "epoch": range(1, len(history.loss_history) + 1),
        "train_loss": history.loss_history,
        "val_loss": history.validation_loss_history,
    },
)
print(train_val_loss_df)
```
